components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -124,6 +126,15 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                 authenticationContext.addParameter(OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
                 authenticationContext.addParameter(OAUTH2_VALIDATE_SCOPE,
                         AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                try {
+                    if (MDC.get(SERVICE_PROVIDER) != null) {
+                        MDC.put(SERVICE_PROVIDER,
+                                OAuth2Util.getServiceProvider(clientApplicationDTO.getConsumerKey())
+                                        .getApplicationName());
+                    }
+                } catch (IdentityOAuth2Exception e) {
+                    log.error("Error occurred while getting the Service Provider by Consumer key");
```

Comment:
Why do we need to swallow the stacktrace here. Don't we loose important information here  ?

Reply:
Fixed with this pr https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/134